python/pyarrow/_dataset.pyx

```diff
@@ -1998,6 +1998,10 @@ cdef class PartitioningFactory(_Weakrefable):
     cdef inline shared_ptr[CPartitioningFactory] unwrap(self):
         return self.wrapped
 
+    @property
+    def type_name(self):
+        return frombytes(self.factory.type_name())
+
```

Comment:
Does this provide any real value?  Note that the C++ impl will currently return `"schema"` for `DirectoryPartitioning`

Reply:
This changes don't use it, I left it around based on this comment from Joris -> https://github.com/apache/arrow/pull/11008#pullrequestreview-742596817

We can remove it if you feel it causes any kind of issue.

Reply:
That was assuming it would return a useful value. If that's not the case, then maybe let's remove it. 
(do we know what's the reason for "schema"? Because DirectoryPartitioning is created with a schema? We could also change the return value in C++)

Reply:
> We could also change the return value in C++

https://github.com/apache/arrow/pull/11172